### PR TITLE
Fix a few bugs and add compatibilty with mac

### PR DESCRIPTION
**Pull request by torognes**
I have made some changes that should make it possible to run the tests under macos. There were a few GNU-specific things originally that I have removed. There were also a few bugs that did not work on Linux either. I have tested it a bit and at least I get the same results when I run it on Linux and Mac with the current vsearch version, but I cannot guarantee that it works correctly in all cases.

The BSD-based unix on Macs lacks some GNU-specific extensions like the Perl-style regular expression in with grep (-P) (also indicated as highly experimental on Linux). Upper and lower case conversion with "^^" and ",," in bash are not supported. The md5sum (md5), sha1sum (shasum) and wc commands are slightly different.

I think there was a problem with the testing of the fastx_subsample command which I have now fixed.

### CODE REVIEW

**Comment by frederic-mahe** [comment]
Excellent work! Now the test suite can run on more systems.

File: scripts/fastx_subsample.sh [hunk 1]
```diff
@@ -897,7 +900,7 @@ OUTPUT=$(mktemp)
 printf "@s1\nA\n+\nG\n@s2\nA\n+\nG\n" | \
     "${VSEARCH}" --fastx_subsample - --fastqout "${OUTPUT}" --sample_size 2 \
 	&> /dev/null
-    [[ $(wc -l < "${OUTPUT}") == "8" ]] && \
+    [[ $(echo $(wc -l < "${OUTPUT}")) == "8" ]] && \
```
Comment (frederic-mahe) [inline scripts/fastx_subsample.sh:903]: I can rewrite the test in a more concise way to remove the need for a storage variable and a tmp file:
```sh
printf "@s1\nA\n+\nG\n@s2\nA\n+\nG\n" | \
    vsearch \
        --fastx_subsample - \
        --sample_size 2 \
        --fastqout - 2> /dev/null | \
    awk 'END {exit NR == 8 ? 0 : 1}' && \
    echo "success" || \
        echo "failure"  ## vsearch's output should be 8 lines of fastq
```
Does it work for you on MacOS @torognes ?

**Comment by frederic-mahe** [comment]
Hi @torognes, many tests were written by two students and need reviewing.

I've been focusing on the dereplication command this afternoon, and it seems that many tests could be refactored (simpler structure, no temporary file, less intermediate variables). To do that, I rely on `grep -q` and `awk {exit condition ? 0 : 1}`. If both these structures work on MacOS, then I shall continue.

I also noticed missing tests. A lot of work in perspective.

**Comment by frederic-mahe** [comment]
Hi @torognes, many tests were written by two students and need reviewing.

I've been focusing on the dereplication command this afternoon, and it seems that many tests could be refactored (simpler structure, no temporary file, less intermediate variables). To do that, I rely on `grep -q` and `awk {exit condition ? 0 : 1}`. If both these structures work on MacOS, then I shall continue.

I also noticed missing tests. A lot of work in perspective.

File: scripts/fastx_subsample.sh (hunk 1 above)
Comment (torognes) [reply scripts/fastx_subsample.sh:903]: I have tested this and these changes work well.

**Comment by torognes** [comment]
It seems like `grep` (except for `-P`) and `awk` work well and identical on macos, so using these constructs should be fine.

**Comment by frederic-mahe** [comment]
Awesome! using these constructs should allow us to avoid a lot of tmp files and storage variables. However, `grep -P` will remain necessary to test the SAM format regexes, for now.

**Comment by torognes** [comment]
Would it be possible to use `grep -e` or `grep -E` instead of `grep -P`? These options also allows for (extended) regular expressions, though not Perl-style.

**Comment by frederic-mahe** [comment]
I would have to review the regex syntax to replace the perl-specific parts, but it might be possible. It works with your corrections though, so this is not a priority.

**Pull request merged**